pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -404,11 +439,14 @@ func (r *ReconcileBuildRun) createTaskRun(ctx context.Context, build *buildv1alp
 		return nil, handleError(fmt.Sprintf("Unsupported BuildStrategy Kind: %v", build.Spec.StrategyRef.Kind), err, updateErr)
 	}
 
+	// Commented out. This code was not being used before. The latest code will use it because there is
+	// an UPDATE call after this lines of code. However, this will conflict with the finalizer
+	// implementation on the Build. See https://github.com/redhat-developer/build/issues/324 for more details.
 	// Set OwnerReference for Build and BuildRun
-	if err := r.setOwnerReferenceFunc(build, buildRun, r.scheme); err != nil {
-		updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
-		return nil, handleError("Failed to set OwnerReference for Build and BuildRun", err, updateErr)
-	}
+	// if err := r.setOwnerReferenceFunc(build, buildRun, r.scheme); err != nil {
```

Comment:
Why we want to comment this `setOwnerReferenceFunc` ? If we don't use it why we do not remove it directly?

Reply:
I wanted to highlight some of the history behind that function. So its fine to keep it there, I believe.

Reply:
Also +1 from me to remove the code.

Reply:
done